run.sh

```diff
@@ -191,14 +191,45 @@ generate_distribution_load() {
 
 get_rps() {
     closed_loop_result_dir=$(echo "${result_dir}" | sed "s|-${distribution}|-closed|g")
+    closed_loop_result_dir=$(echo "${closed_loop_result_dir}" | sed "s|-normalized||g")
     tput_csv=${closed_loop_result_dir}/tput.csv
+    num_cols=$(awk -F',' '{print NF; exit}' ${tput_csv})
 
+    # Declare a multipler
+    declare -a max_multiplier
+    for ((i = 0; i < $((num_cols-2)); i++)); do
+        max_multiplier[$i]=1
+    done
+
+    # Normalize the throughput if needed
+    if [[ ${is_normalized} == true ]]; then
+        while IFS=, read -r mode load rest; do
+            IFS=, read -ra cols <<< "$rest"
+            i=0
+            for col in "${cols[@]}"; do
+                if (( $(echo "$col > ${max_multiplier[$i]}" | bc -l) )); then
+                    max_multiplier[$i]=$col
+                fi
+                i=$((i+1))
+            done
+        done < <(tail -n +2 "${tput_csv}")
+
+        for ((i = 0; i < ${#max_multiplier[@]}; i++)); do
+            element=${max_multiplier[$i]}
+            reciprocal=$(echo "scale=10; 1/$element" | bc -l)
+            max_multiplier[$i]=$reciprocal
+        done
+    fi
+
+    # Find max (normalized) tput
     max_sum=0
     while IFS=, read -r mode load rest; do
         IFS=, read -ra cols <<< "$rest"
         sum=0
+        i=0
         for col in "${cols[@]}"; do
-            sum=$(echo "$sum + $col" | bc)
+            sum=$(echo "$sum + ($col * ${max_multiplier[$i]})" | bc)
```

Comment:
If normalized is not passed, then all values in the array would be 1, and thus would work as aggregate tput